modules/tensor_mechanics/src/actions/TensorMechanicsActionBase.C

```diff
@@ -18,38 +18,71 @@ const std::map<std::string, std::string>
         {"strain", "total_strain"},
         {"stress", "stress"},
         {"elastic_strain", "elastic_strain"},
+        {"elastic_stress", "elastic_stress"},
         {"plastic_strain", "plastic_strain"},
-        {"creep_strain", "creep_strain"}};
+        {"plastic_stress", "plastic_stress"},
+        {"creep_strain", "creep_strain"},
+        {"creep_stress", "creep_stress"}};
 const std::vector<char> TensorMechanicsActionBase::_component_table = {'x', 'y', 'z'};
 
 // map aux variable name prefixes to RankTwoInvariant option and list of permitted tensor name
 // shortcuts
 const std::map<std::string, std::pair<std::string, std::vector<std::string>>>
     TensorMechanicsActionBase::_rank_two_invariant_table = {
-        {"vonmises", {"VonMisesStress", {"stress"}}},
-        {"effective", {"EffectiveStrain", {"plastic_strain", "creep_strain"}}},
-        {"hydrostatic", {"Hydrostatic", {"stress"}}},
-        {"l2norm", {"L2norm", {"stress", "strain"}}},
-        {"volumetric", {"VolumetricStrain", {"strain"}}},
-        {"firstinv", {"FirstInvariant", {"stress", "strain"}}},
-        {"secondinv", {"SecondInvariant", {"stress", "strain"}}},
-        {"thirdinv", {"ThirdInvariant", {"stress", "strain"}}},
-        {"triaxiality", {"TriaxialityStress", {"stress"}}},
-        {"maxshear", {"MaxShear", {"stress"}}},
-        {"intensity", {"StressIntensity", {"stress"}}},
-        {"max_principal", {"MaxPrincipal", {"stress"}}},
-        {"mid_principal", {"MidPrincipal", {"stress"}}},
-        {"min_principal", {"MinPrincipal", {"stress"}}}};
+        {"vonmises",
```

Comment:
Whoa, hold on here!

Not every one of the invariants makes sense for every possible tensor, which is why we had a limited set defined originally. Some of these can be expanded to allow for other quantities to be computed, but not all of them, and we need to do this in a thoughtful way.  For now, please revert all of them back to the original set. 

Right off the top of my head, these are the obvious ones that could be expanded:
l2norm: It's OK for all of them
max_principal, mid_principal, min_principal: These should also be applicable to the strain
If we're going to add any, we need to test to make sure we've got them right, and add regression tests. For now, let's just leave them alone, and come back later and expand it.

Reply:
I set all these conditions to quickly convert over input files in BISON.  Files like RIA_CABRI_REP_Na4 have auxvariable/auxkernel pairs that follow the same convention as the already established sets.  (i.e. axial_creep_strain, hoop_elastic_strain, etc. )  I planned on converting over theses pairs, then removing the unnecessary tensors from the MOOSE PR.  I thought it would be faster to remove the tensors, than to collect all the necessary conditions, rebuild MOOSE, rebuild BISON, and then change the input files.

Reply:
I think those ones you mentioned make sense. Go ahead and submit another PR that adds the specific cases that are being used. I just don't think we can quite go and wholesale add every combination of options.